core/amber/src/main/scala/edu/uci/ics/texera/web/resource/dashboard/workflow/WorkflowExecutionsResource.scala

```diff
@@ -3,33 +3,42 @@ package edu.uci.ics.texera.web.resource.dashboard.workflow
 import edu.uci.ics.texera.web.SqlServer
 import edu.uci.ics.texera.web.auth.SessionUser
 import edu.uci.ics.texera.web.model.jooq.generated.Tables.{WORKFLOW, WORKFLOW_EXECUTIONS}
-import edu.uci.ics.texera.web.resource.dashboard.workflow.WorkflowExecutionsResource.{
-  WorkflowExecutionEntry,
-  context
-}
+import edu.uci.ics.texera.web.model.jooq.generated.tables.daos.WorkflowExecutionsDao
+import edu.uci.ics.texera.web.model.jooq.generated.tables.pojos.WorkflowExecutions
+import edu.uci.ics.texera.web.resource.dashboard.workflow.WorkflowExecutionsResource._
 import io.dropwizard.auth.Auth
 import org.jooq.types.UInteger
 
 import java.sql.Timestamp
 import javax.annotation.security.PermitAll
 import javax.ws.rs._
-import javax.ws.rs.core.MediaType
+import javax.ws.rs.core.{MediaType, Response}
 import scala.collection.convert.ImplicitConversions.`collection AsScalaIterable`
 
 object WorkflowExecutionsResource {
   final private lazy val context = SqlServer.createDSLContext()
+  final private lazy val executionsDao = new WorkflowExecutionsDao(context.configuration)
+
+  def getExecutionById(eId: UInteger): Option[WorkflowExecutions] = {
```

Comment:
since the request is coming from the list itself not an outside source, it is better to not make it option

Reply:
But it is an option because an execution might not exist by that eid? How else would you convey failure? Is an exception better?

Reply:
since the eid is actually chosen from the list, we can safely assume it exists. also try to use `fetchOneByEid` if such a method exists to fetch a single record since eid is unique